### PR TITLE
Draft: Profile 1.0 Spec

### PR DESCRIPTION
**Pull request by mlagally**
This branch contains a proposal of how the profile 1.0 release can be structured. 

Scope has been reduced to only normatively define the HTTP baseline profile. It makes the two event profiles optional.
Language has been cleaned up, the introduction section has been improved, the two event profiles were made informative and corresponding language has been provided.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-profile/pull/272.html" title="Last updated on Sep 5, 2022, 11:34 AM UTC (1abd758)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-profile/272/f7348cf...1abd758.html" title="Last updated on Sep 5, 2022, 11:34 AM UTC (1abd758)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -171,8 +177,9 @@
       other features of the Thing Description that go beyond the constraints
       of the HTTP Baseline Profile, however the interoperability guarantees of the HTTP Baseline Profile
       hold only for the <a>HTTP Baseline Profile</a> subset.
-
-
+    </p>
+    <p>Future versions of this document may contain other profiles,
+      e.g. a profile for digital twins and a profile for resource constrained devices.</p>
```
Comment (benfrancis) [inline index.html:182]: I would prefer not to list digital twins as an example because I'm not yet convinced that's a good example use case for a profile. Better to just use the constrained devices example which is less contentious.

File: index.html [hunk 2]
```diff
@@ -759,796 +678,43 @@ <h1>Common Constraints</h1>
...
+      </p>
+      <p>
+        As described by this section the following constraints must be observed:
+        <span class="rfc2119-assertion" id="common-constraints-date-format-2">
+          All dateTime values SHOULD use UTC as the time zone and use the 'Z' identifier.</span>
+      </p>
+      <p>
+        <span class="rfc2119-assertion" id="common-constraints-date-format-3">
+          A time value of 24:00 is not permitted; the value 00:00 MUST be used instead.</span>
+      </p>
```
Comment (benfrancis) [inline index.html:717]: In specref, [[xmlschema-2]] points to https://www.w3.org/TR/xmlschema-2/ which doesn't appear to have a section 3.2.7.

That spec does say _"'24' is permitted if the minutes and seconds represented are zero"_, which is not what you want.

In https://github.com/w3c/wot-profile/pull/267 I proposed using the reference [[RFC3339]] (https://www.rfc-editor.org/rfc/rfc3339) instead, since that already says:

_"Although ISO 8601 permits the hour to be "24", this profile of ISO 8601 only allows values between "00" and "23" for the hour in order to reduce confusion."_

RFC 3339 also only supports UTC and only numerical offsets (not those that don't have an integral number of minutes). I believe this resolves the ambiguity of timezones you are concerned about, without being quite so prescriptive.

In short, if you just reference RFC3339, the rest of the qualifying text is not needed.

File: index.html [hunk 3]
```diff
@@ -759,796 +678,43 @@ <h1>Common Constraints</h1>
       The following sections are applicable for all of the HTTP profiles defined by this document.
     </p>
 
-    <section id="http-profiles-information-model">
-      <h2>Information Model</h2>
```
Comment (benfrancis) [inline index.html:763]: 🎉

**Review by benfrancis — COMMENTED** [review]
There are two separate large changes in this PR...

The first appears to be an alternative to https://github.com/w3c/wot-profile/pull/267 and replaces the Information Model section with separate smaller sections which are less contentious. I strongly support this change, and actually prefer it to my own PR.

The second is to make the HTTP SEE Profile and HTTP Webhook Profile informative. I think that requires further discussion.

File: index.html (hunk 1 above)
Comment (mlagally) [reply index.html:182]: Will change to "profiles for other protocols and profiles for resource constrained devices."

Comment (mlagally) [reply index.html:182]: Profile call on Wed 7h: remove lines 180-182

**Comment by mlagally** [comment]
Agreed to merge in Profile call on Sept 7th.

File: index.html (hunk 2 above)
Comment (mlagally) [reply index.html:717]: This is text we previously had agreed on. If it needs to be improved, let's treat it in a separate issue.
I filed a new issue for a follow up.
